### PR TITLE
CI: Use install_go.sh  script

### PR DESCRIPTION
**Pull request by jcvenegas**
Check if go is installed.

 - If go requested version is installed, then do nothing.

 - If go is installed with other version, fail requesting remove it first.

- Clone repository using git
- Install go

### CODE REVIEW

File: .ci/jenkins_job_build.sh [hunk 1]
```diff
@@ -33,17 +37,18 @@ mkdir -p "${GOPATH}"
...
 
-# Get the tests repo in case this repo was cloned using
-# `git clone` and it is not in the $GOPATH
-if [ "${kata_repo}" != "${tests_repo}" ]
-then
-	go get -u -d "${tests_repo}" || true
-fi
+# Get the tests repository
+mkdir -p $(dirname "${tests_repo_dir}")
+[ -d "${kata_repo_dir}" ] || git clone "https://${tests_repo}.git" "${tests_repo_dir}"
```
Comment (chavafg) [inline .ci/jenkins_job_build.sh]: `[ -d "${tests_repo_dir}" ]` ?

Comment (jcvenegas) [reply .ci/jenkins_job_build.sh]: done!

**Comment by jcvenegas** [comment]
@chavafg  the CI is faling as  expected:
we need to merge #247, then remove the go installation from the each azure job.
ERROR: go version go1.9.2 linux/amd64 is installed, remove it before install go 1.10

File: .ci/install_go.sh [hunk 2]
```diff
@@ -81,6 +81,13 @@ if [ -z "$go_version" ];then
 	usage 0
 fi
 
+if command -vV go; then
```
Comment (jodh-intel) [inline .ci/install_go.sh]: Only one of `-v` or `-V` is required fwics?

File: .ci/jenkins_job_build.sh [hunk 3]
```diff
@@ -13,6 +13,10 @@ set -e
 # Need the repo to know which tests to run.
 export kata_repo="$1"
 
+echo "Setup env for kata repository: $kata_repo"
+
+[ -z "$kata_repo" ] && echo "kata repo no provided" && exit 1
```
Comment (jodh-intel) [inline .ci/jenkins_job_build.sh]: `echo >&2 "kata repo no provided"` ?

**Comment by chavafg** [comment]
@jcvenegas #247 is merged. Once you address @jodh-intel comments I'll remove the `golang` installation from the VM setup and see if this PR works as expected. I'll do it on the Fedora Job, since it is working only for this repository (don't want to have a lot of jobs failing). If everything goes well, we can merge and remove the golang installation from all other distros.

**Comment by jcvenegas** [comment]
@jodh-intel  @chavafg  branch updated. Ready to test without golang in azure jobs.

**Comment by jodh-intel** [comment]
lgtm

**Comment by chavafg** [comment]
lgtm

**Comment by chavafg** [comment]
adding dnm as there were some errors on the centos and ubuntu 17.10 CI

**Comment by jcvenegas** [comment]
Do no merge again, moving to go 1.10 has some issues in virtcotainerd (linter and unit test).

**Comment by jcvenegas** [comment]
now failing `package manager update test check yum update [It] should not fail` not related with this PR

**Comment by GabyCT** [comment]
@jcvenegas , I will re-try this as a timeout appear as an error

**Comment by jcvenegas** [comment]
@GabyCT  @jodh-intel @chavafg @grahamwhaley ready to merge all green 🥁

**Comment by jodh-intel** [comment]
I :heart: green buttons! :grinning:

**Comment by grahamwhaley** [comment]
mumble - retrospectively - the 'if a different version is installed, abort' action is a mild pain for the metrics CI actually - we use 'go' in the Jenkins master-side script to grab and install some stuff, so need some version installed in the build slave....
I can make a temporary workaround I think on the Jenkins side - and sorry I didn't pick up on this at the time, but the metrics CIs were down for other reasons (which I'm working on now).

I wonder if we can't just install the go versions in different paths, and not have a one 'system level' go - and then work with that. Maybe I'll PR that if that turns out to be the nice way forwards.

**Pull request merged**